tests/suite-unit/PluginFlyvemdmAgent.php

```diff
@@ -348,4 +384,163 @@ public function testGetSpecificValueToDisplay() {
+            'input' => ['_unenroll_request' => 'lorem'],
+            'expected' => [
+               'result' => [
+                  '_unenroll_request' => 'lorem',
+                  'enroll_status'     => 'unenrolling',
+               ],
+            ],
+         ],
+         'agent response for inventory' => [
+            'input'    => ['_inventory' => $agentXmlInventory],
+            'expected' => [
+               'result' => [
+                  '_inventory'   => $agentXmlInventory,
+                  'last_contact' => $_SESSION['glpi_currenttime'],
+               ],
+            ],
+            'extra'    => ['isAgent' => true],
+         ],
+         'agent response for online status' => [
+            'input'    => ['is_online' => true],
+            'expected' => [
+               'result' => [
+                  'is_online'    => '1',
+                  'last_contact' => $_SESSION['glpi_currenttime'],
+               ],
+            ],
+            'extra'    => ['isAgent' => true],
+         ],
+         'agent response for offline status' => [
+            'input'    => ['is_online' => false],
+            'expected' => [
+               'result' => [
+                  'is_online'    => '0',
+                  'last_contact' => $_SESSION['glpi_currenttime'],
+               ],
+            ],
+            'extra'    => ['isAgent' => true],
+         ],
+         'agent response for invalid status' => [
+            'input'    => ['is_online' => 'lorem'],
+            'expected' => ['result' => false, 'message' => 'Invalid status value'],
+            'extra'    => ['isAgent' => true],
+         ],
+      ];
+   }
+
+   /**
+    * @dataProvider providerUpdateInput
+    * @tags testGetprepareInputForUpdate
+    *
+    * @param array $input
+    * @param array $expected
+    * @param array $extra
+    */
+   public function testGetprepareInputForUpdate(array $input, array $expected, array $extra = []) {
+      $instance = $this->newMockInstance(\PluginFlyvemdmAgent::class);
+      $instance->fields['plugin_flyvemdm_fleets_id'] = isset($extra['mockFleet']) ? $extra['mockFleet'] : null;
+      $instance->fields['wipe'] = isset($extra['mockWipe']) ? $extra['mockWipe'] : 0;
+      if (isset($extra['isAgent']) && $extra['isAgent']) {
+         $config = \Config::getConfigurationValues('flyvemdm', ['agent_profiles_id']);
+         $_SESSION['glpiactiveprofile']['id'] = $config['agent_profiles_id'];
+      } else {
+         $_SESSION['glpiactiveprofile']['id'] = 1;
+      }
+      $instance->getMockController()->update = true;
+      $instance->getMockController()->getTopic = isset($extra['mockTopic']) ? $extra['mockTopic'] : null;
+      $instance->getMockController()->notify = null;
+      $result = $instance->prepareInputForUpdate($input);
+      if ($expected['result'] === false) {
+         $this->boolean($result)->isFalse();
+         $this->string($_SESSION["MESSAGE_AFTER_REDIRECT"][0][0])->isEqualTo($expected['message']);
```

Comment:
you built a few months ago a methos to check messages in $_SESSION, use it to not rely on an index